### PR TITLE
Searchbar fixes

### LINKED ISSUE #3349: Bug: auto-suggestion search bar not working properly when using keyboard navigation keys
branch - develop 

Current behaviour - The auto-suggestion search bar not working properly, when using keyboard   navigation keys. I think it displays the name of exercise which is saved in the local directory.

Expected behaviour - It should display the name of exercise which is given on the web. And should navigate to the correct result.

As you can see in the screenshot,  name of the exercise is "Introduction: exploring college options" but It doesn't shows the same thing above.  

screen shot 2015-03-20 at 12 28 33 am png

Another screenshot:
screen shot 2015-03-20 at 12 50 49 am png

Screenshot of buggy result:
screen shot 2015-03-20 at 12 30 42 am

### LINKED ISSUE #3578: Search autocomplete is ugly and twitchy
Something weird has happened to the styling of search autocomplete between 0.13 and develop.

This may have been my fault.

Any CSS wizards available to hunt down where the styling has gone?

Also, I think this has always been the case, but it is twitchy when you mouse over it.

@jtamiace @Antrikshy?

### PR DESCRIPTION
**Pull request by MCGallaspy**
Fixes #3578. I couldn't figure out what happened to the `0.13.x` styling, so I just tried my own. Perhaps the only contentious thing here is that search result elements have a fixed height; overflow is hidden, instead of increasing the result item's height.

Also fixes #3349 and #3345.

Screenshot:
ss

### CODE REVIEW

**Comment by rtibbles** [comment]
Yeah, I was so mystified by what happened to the 0.13.x styling.

**Comment by rtibbles** [comment]
One thing that comes to mind - could we sort first by availability in the search?

**Comment by rtibbles** [comment]
(feel free to tell me to sling my hook and make an issue)

**Comment by MCGallaspy** [comment]
We can and we will!

File: kalite/distributed/static/css/distributed/search_autocomplete.css [hunk 1]
```diff
@@ -29,26 +29,40 @@
...
+}
+
 .navbar-collapse .ui-menu .ui-menu-item span {
     font-weight: 300;
     font-size: 0.95em;
     padding-right: 0;
 }
 
+/* inner spans */
+.navbar-collapse .ui-menu .ui-menu-item span > span {
```
Comment (rtibbles) [inline kalite/distributed/static/css/distributed/search_autocomplete.css]: Much specific selector.

File: kalite/distributed/static/js/distributed/search/views.js [hunk 2]
```diff
@@ -116,6 +116,14 @@ window.AutoCompleteView = BaseView.extend({
             return compvalue;
         });
 
+        // Then sort again by availability!
+        ids_filtered.sort(function(id1, id2) {
```
Comment (rtibbles) [inline kalite/distributed/static/js/distributed/search/views.js]: Hrm - it seems that this sort destroys the order imposed by the previous sort?

Topic sandwiched between two exercises here:

image

Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/search/views.js]: Right, the stability of Array.prototype.sort depends on the browser. Will use a stable sort.

**Comment by rtibbles** [comment]
Order is restored!
image

**Pull request merged**